model/action_test.go

```diff
@@ -0,0 +1,551 @@
+package model
+
+import (
+	"github.com/stretchr/testify/require"
+	"github.com/unidoc/unipdf/v3/core"
+	"strings"
+	"testing"
+)
+
+func testAction(t *testing.T, rawText string, actionType PdfActionType, testFunc func(t *testing.T, action *PdfAction)) {
```

Comment:
A short description would be nice. Seems to parse the action as object number 1 from `rawText` and check that it matches the `actionType`.  Then it applies testFunc() on the action (action-specific?) and checks finally checks that the serialized output is the same as the input rawText.